packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-preview.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useMemo } from 'react';
+import { connect } from 'react-redux';
+import {
+  Body,
+  css,
+  SpinLoader,
+  DocumentIcon,
+  spacing,
+  Overline,
+} from '@mongodb-js/compass-components';
+import type { RootState } from '../../../modules';
+import type { Document } from 'mongodb';
+import { DocumentListView } from '@mongodb-js/compass-crud';
+import HadronDocument from 'hadron-document';
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  height: '100%',
+});
+
+const previewHeaderStyles = css({
+  paddingTop: spacing[3],
+  paddingBottom: spacing[3],
+  paddingLeft: spacing[3],
+});
+
+const centerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  justifyContent: 'center',
+  height: '100%',
+});
+
+const messageStyles = css({ marginTop: spacing[3] });
+
+const documentListStyles = css({
+  overflow: 'scroll',
+});
+
+type PipelineAsTextWorkspaceProps = {
+  isLoading: boolean;
+  previewDocs: Document[] | null;
+  sampleSize: number;
+  isAutoPreview: boolean;
+};
+
+const PreviewHeader = ({ sampleSize }: { sampleSize: number }) => {
+  return (
+    <div className={previewHeaderStyles}>
+      <Overline>Pipeline Output</Overline>
+      <Body>{`Sample of ${sampleSize} documents`}</Body>
+    </div>
+  );
+};
+
+const PreviewResults = ({
+  previewDocs,
+  isLoading,
+  isAutoPreivew,
+}: {
+  previewDocs: Document[] | null;
+  isLoading: boolean;
+  isAutoPreivew: boolean;
+}) => {
+  const listProps: React.ComponentProps<typeof DocumentListView> = useMemo(
+    () => ({
+      docs: (previewDocs ?? []).map((doc) => new HadronDocument(doc)),
+      isEditable: false,
+      copyToClipboard(doc) {
+        const str = doc.toEJSON();
+        void navigator.clipboard.writeText(str);
+      },
+    }),
+    [previewDocs]
+  );
+
+  if (!isAutoPreivew) {
+    return (
+      <div className={centerStyles}>
+        <DocumentIcon size={spacing[6]} />
+        <Body className={messageStyles}>
+          Preview results to see a sample of the aggregated results from this
+          pipeline.
+        </Body>
+      </div>
+    );
+  }
```

Comment:
Shouldn't we completely hide this panel when autopreview is off like we do in stage by stage editor?